File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0;
        int r = nums.size() - 1;

        while(l < r){
            int middle = (l + r) / 2;

            if(nums[middle] > nums[r]){
                l = middle + 1;
            } else if(nums[middle] <= nums[r]){
                r = middle;
            }
        }
        return nums[l];
    }
};

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        if sum(nums) % 2:
            return False

        target = sum(nums) // 2
        dp = set()
        dp.add(0)

        for num in nums:
            nextDP = set()
            for currSum in dp:
                if target in dp:
                    return True
                nextDP.add(currSum)
                nextDP.add(currSum + num)
            dp = nextDP
        
        return True if target in dp else False